egui/src/containers/frame.rs

```diff
@@ -3,12 +3,49 @@
 use crate::{layers::ShapeIdx, *};
 use epaint::*;
 
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct Margin {
+    pub left: f32,
+    pub right: f32,
+    pub top: f32,
+    pub bottom: f32,
+}
+
+impl Margin {
+    /// Margins with the same size on opposing sides
+    #[inline]
+    pub fn symmetric(x: f32, y: f32) -> Self {
+        Self {
+            left: x,
+            right: x,
+            top: y,
+            bottom: y,
+        }
+    }
+
+    /// Total margin in x direction
+    pub fn x(&self) -> f32 {
+        self.left + self.right
+    }
+
+    /// Total margin in y direction
+    pub fn y(&self) -> f32 {
```

Comment:
If I read `margin.y()` I wouldn't guess it is the sum of the top and bottom margins. Perhaps instead we could do `fn sum(&self) -> Vec2` so users could do `margin.sum().y` for the same effect?